react/components/NumericStepper/index.js

```diff
@@ -0,0 +1,223 @@
+import PropTypes from 'prop-types'
+import React from 'react'
+
+const normalizeMin = min => min == null ? -Infinity : min
+const normalizeMax = max => max == null ? Infinity : max
+
+const validateValue = (value, min, max, defaultValue) => {
+  min = normalizeMin(min)
+  max = normalizeMax(max)
+
+  if (isNaN(value) || value == null) {
+    if (defaultValue < min) return min
+    if (defaultValue > max) return max
+    return defaultValue
+  } else if (value < min) {
+    return min
+  } else if (value > max) {
+    return max
+  }
+  return parseInt(value, 10)
+}
+
+const validateDisplayValue = (value, min, max) => {
```

Comment:
Qual a diferenca dessa funcao pra `validateValue`? Elas sao praticamentes iguais, mesma entrada, mesma saida.

Reply:
@klzns A diferença eu comentei no state do displayValue, talvez caiba comentar na função tambem: ela é pra estados temporariamente inválidos de valor durante a digitação (especificamente, string vazia e sinal de negativo sem numero). Durante a digitação, o valor interno é validado e resulta em um número, mas o valor de display mostra o que tá sendo digitado.

Reply:
Droga, meu comentario nao foi.

E se colocar o minimo como 10 e o maximo como 25.

O usuario digitar "1" e enquanto ele vai digitar o proximo digito "2", vc ve que 1 é menor que 10 e troca o numero para 10.

Isso acontece?

Reply:
@klzns Bem pensado, acho que nao funciona nao.

Essa lógica funciona bem pra valores máximos e quando o mínimo for 0 ou 1, o que é o caso mais comum (por exemplo, quantidade de produtos). Acho que vou tratar esses como casos especiais.